vllm/model_executor/layers/quantization/torchao.py

```diff
@@ -0,0 +1,179 @@
+# SPDX-License-Identifier: Apache-2.0
+from typing import Any, Dict, List, Optional
+
+import torch
+import torch.nn.functional as F
+
+from vllm.model_executor.layers.linear import LinearBase
+from vllm.model_executor.layers.quantization.base_config import (
+    QuantizationConfig, QuantizeMethodBase)
+from vllm.model_executor.parameter import ModelWeightParameter
+
+
+class TorchAOMethod(QuantizeMethodBase):
+    """Implements quantization method for TorchAO."""
+
+    def __init__(self, config):
+        self.config = config
+        self.weight_bits = (config.bit_width
+                            )  # Changed from weight_bits to bit_width
+        self.group_size = (
+            128  # Added default group size, should be configurable
+        )
+        self.scheme = config.scheme
+
+    def create_weights(
+        self,
+        layer: torch.nn.Module,
+        input_size_per_partition: int,
+        output_partition_sizes: List[int],
+        input_size: int,
+        output_size: int,
+        params_dtype: torch.dtype,
+        **extra_weight_attrs,
+    ):
+        """Create quantized weights for the layer."""
+        del input_size, output_size
+        output_size_per_partition = sum(output_partition_sizes)
+        weight_loader = extra_weight_attrs.get("weight_loader")
+        layer.logical_widths = output_partition_sizes
+        layer.input_size_per_partition = input_size_per_partition
+        layer.output_size_per_partition = output_size_per_partition
+        # weight_dtype = (
+        #     torch.float8_e4m3fn
+        #     if self.quant_config.is_checkpoint_fp8_serialized
+        #     else params_dtype
+        # )
+        # torch.distributed.breakpoint()
+        from torchao.quantization.quant_api import quantize_,Int8WeightOnlyConfig
+
+
+        weight_dtype = layer.params_dtype
+
+        tst = torch.nn.Linear(input_size_per_partition, output_size_per_partition, bias=False)
+        quantize_(tst, Int8WeightOnlyConfig())
```

Comment:
It seems like all methods create the empty param and then later on the weight_loader copies into them this feels like the pivotal gap since AQT doesn't support copy_